stoppable_workers.go

```diff
@@ -68,6 +70,28 @@ func (sw *StoppableWorkers) Add(worker func(context.Context)) {
 	})
 }
 
+// StartTimer is an optional call that creates a timer that can be used in conjunction with
+// `NextTick`.
+func (sw *StoppableWorkers) StartTimer(duration time.Duration) {
+	sw.timer = time.NewTimer(duration)
+}
+
+// NextTick blocks until the timer ticks or the StoppableWorkers has been canceled. It returns true
+// if the timer has ticked and false if the context is canceled. Such that one can write a loop to
+// do work as such:
+//
+//	for sw.NextTick() {
+//	  doWork()
+//	}
+func (sw *StoppableWorkers) NextTick() bool {
+	select {
+	case <-sw.timer.C:
```

Comment:
I would guess `NextTick()` before `StartTimer` could panic due to accessing `.C`.

Reply:
Correct -- also talked offline and also leveraging the type system to prevent this programmer error.